### PR TITLE
test(precompiles-macros): add edge-case unit tests for allocate_slots

### PR DESCRIPTION
**Pull request by decofe**
Add edge-case unit tests for the packing module core functions: allocate_slots, classify_field_type, PackingConstants, and SlotAssignment. Covers empty inputs, auto/manual/base_slot assignment chains, interleaved slot modes, mapping vs direct type classification, constant naming, and boundary values.

### CODE REVIEW

**Review by 0xrusowsky — APPROVED** [review]

File: crates/precompiles-macros/src/packing.rs [hunk 1]
```diff
@@ -453,3 +453,241 @@ pub(crate) fn gen_collision_check_fn(
...
+        assert!(matches!(
+            classify_field_type(&parse_quote!(Option<u32>)).unwrap(),
+            FieldKind::Direct(_)
+        ));
+
+        // simple mapping → Mapping
+        assert!(matches!(
+            classify_field_type(&parse_quote!(Mapping<Address, U256>)).unwrap(),
+            FieldKind::Mapping { .. }
+        ));
```
Comment (0xrusowsky) [inline crates/precompiles-macros/src/packing.rs]: nit: we can probably use a closure to make it more succinct

File: crates/precompiles-macros/src/packing.rs [hunk 2]
```diff
@@ -453,3 +453,241 @@ pub(crate) fn gen_collision_check_fn(
...
+            .ref_slot(),
+            U256::ZERO
+        );
+        assert_eq!(
+            *SlotAssignment::Auto {
+                base_slot: U256::from(u64::MAX)
+            }
+            .ref_slot(),
+            U256::from(u64::MAX)
+        );
```
Comment (0xrusowsky) [inline crates/precompiles-macros/src/packing.rs]: same here

File: crates/precompiles-macros/src/packing.rs [hunk 3]
```diff
@@ -453,3 +453,241 @@ pub(crate) fn gen_collision_check_fn(
...
+    }
+
+    use crate::utils::extract_mapping_types;
```
Comment (0xrusowsky) [inline crates/precompiles-macros/src/packing.rs]: nit: import at the top
